More infrastructure to bring EOS desktop back to life

mariospr opened the pull request:
This is a list of commits that I are needed to bring the desktop back to life and that, most of them, will be needed later on to implement further changes that are at the moment living in my WIP branch here: https://github.com/endlessm/gnome-shell/commits/wip/mario/icongrid

The main reason I'm proposing this PR now is because I will be away until June 4th and this way it should be easier for others to either resume the work I have in my WIP branch (where I have a more functional desktop, but still in a very rudimentary state) or to work on other tasks that can also depend on some of the bits contained here (e.g. implementing the "Endless button" requires the commit that adds `workspaceMonitor.js, that's why I put it first in the llist).

The other reasons for me to propose this specific patch set and not to include other commits from my WIP branch is because the commits suggested here should be fairly safe to integrate in master, since the most visible change is the removal of the "Frequent / All" selector in the apps mode in the overview, and the fact that such view now loads the icons & folders from the Endless desktop, instead of the system ones.

@cosimoc: Please review. If you do it before this Monday there's a small chance I can look into addressing comments (no promises though), otherwise feel free to adjust the commits yourself as you see fit. Thanks!

https://phabricator.endlessm.com/T17230

cosimoc commented on js/ui/appDisplay.js:
```diff
@@ -1089,6 +1116,13 @@ const AppSearchProvider = new Lang.Class({
         return new AppIcon(app);
+    },
+
+    activateResult: function(appId) {
```
I think the idea here is that the existing `activate()` method on `AppIcon` will be called no? Not sure you need this method.

mariospr replied:
I took that from the older codebase, where there was no AppIcon being created. I'll remove this method

cosimoc commented on js/misc/util.js:
```diff
@@ -398,3 +425,84 @@ function ensureActorVisibleInScrollView(scrollView, actor) {
+    let root = parser.get_root().get_object();
+
+    let searchProviderDataNode = root.get_member('default_search_provider_data');
+    if (!searchProviderDataNode || searchProviderDataNode.get_node_type() != Json.NodeType.OBJECT) {
```
Remove curly braces for single-line if (many occurrences in this commit)

mariospr replied:
Ok

cosimoc commented on js/ui/workspaceMonitor.js:
```diff
@@ -0,0 +1,71 @@
+                // We do not count transient windows because of an issue with Audacity
+                // where a transient window was always being counted as visible even
+                // though it was minimized
+                if (window.get_transient_for()) {
```
Remove curly braces for single-line if (many occurrences in this commit)

mariospr commented on data/gnome-shell-theme.gresource.xml:
```diff
@@ -48,5 +48,27 @@
     <file>hot-corner-rtl-symbolic.svg</file>
     <file>mini-icon-active-indicator.png</file>
     <file>system-logout.png</file>
+
```
Actually, this commit shouldn't have made it to this PR, it's part of my WIP branch. I'll remove it

mariospr commented on js/ui/backgroundMenu.js:
```diff
@@ -18,8 +19,20 @@ const BackgroundMenu = new Lang.Class({
-        this.addSettingsAction(_("Display Settings"), 'gnome-display-panel.desktop');
-        this.addSettingsAction(_("Settings"), 'gnome-control-center.desktop');
+
+        this.addAction(_("Add App"), Lang.bind(this, function() {
```
I don't think that's a good idea. `addSettingsAction()` is meant to be used for panels of gnome control center, and I don't think it fits well for opening something else like GS. Unless I'm missing something obvious I think that using `addAction()`, either with or without the app activation context, is probably the right thing to do.

cosimoc replied:
OK that makes sense, as long as we're using the speedwagon.

mariospr commented on src/shell-app-system.c:
```diff
@@ -156,15 +215,99 @@ stale_app_remove_func (gpointer key,
+
+      if (app_is_stale (app))
+        {
+          // App is stale, we remove it
```
Ok (not mine, though)

mariospr commented on src/shell-app-system.c:
```diff
@@ -282,6 +443,39 @@ shell_app_system_lookup_heuristic_basename (ShellAppSystem *system,
+  ShellApp *result;
+  const char *id;
+
+  if (alias == NULL)
```
Makes  sense

mariospr commented on src/shell-app-system.c:
```diff
@@ -364,20 +565,45 @@ _shell_app_system_notify_app_state_changed (ShellAppSystem *self,
+  gchar *app_address = g_strdup_printf ("%p", app);
+  GDesktopAppInfo *app_info = shell_app_get_app_info (app);
+  const gchar *app_info_id = NULL;
+  if (app_info != NULL) {
```
Ok

mariospr commented on src/shell-app-system.c:
```diff
@@ -364,20 +565,45 @@ _shell_app_system_notify_app_state_changed (ShellAppSystem *self,
     case SHELL_APP_STATE_STOPPED:
       g_hash_table_remove (self->priv->running_apps, app);
+
+      if (g_hash_table_remove (self->priv->running_apps, app) && app_info_id != NULL)
```
Good catch. That's a bug we currently have too

mariospr commented on configure.ac:
```diff
@@ -104,7 +104,8 @@ SHARED_PCS="gio-unix-2.0 >= $GIO_MIN_VERSION
             gobject-introspection-1.0 >= $GOBJECT_INTROSPECTION_MIN_VERSION
```
Very true, sorry for the sloppy catch-it-all commit

mariospr commented on src/st/st-entry.c:
```diff
@@ -1172,6 +1384,8 @@ _st_entry_set_icon (StEntry       *entry,
+      g_signal_connect (*icon, "button-press-event",
+                        G_CALLBACK (_st_entry_icon_press_cb), entry);
       g_signal_connect (*icon, "button-release-event",
                         G_CALLBACK (_st_entry_icon_press_cb), entry);
```
Indeed

mariospr commented on src/st/st-entry.c:
```diff
@@ -72,7 +72,10 @@ enum
   PROP_0,
```
Ok

mariospr commented on js/ui/shellEntry.js:
```diff
@@ -121,6 +134,293 @@ const EntryMenu = new Lang.Class({
+    },
+
+    _onCapturedEvent: function(actor, event) {
+        if (event.type() == Clutter.EventType.BUTTON_PRESS) {
```
Ok

mariospr commented:
@cosimoc I think I addressed all your comments with the exception of `addSettingsAction()` (see my comment about that), so this should be ready for another review, and maybe merging

cosimoc commented on src/shell-grid-desaturate-effect.c:
```diff
@@ -39,8 +39,6 @@
 #define SHELL_IS_GRID_DESATURATE_EFFECT_CLASS(klass)     (G_TYPE_CHECK_CLASS_TYPE ((klass), SHELL_TYPE_GRID_DESATURATE_EFFECT))
 #define SHELL_GRID_DESATURATE_EFFECT_GET_CLASS(obj)      (G_TYPE_INSTANCE_GET_CLASS ((obj), SHELL_TYPE_GRID_DESATURATE_EFFECT, ShellGridDesaturateEffectClass))
 
-#define CLUTTER_ENABLE_EXPERIMENTAL_API
```
Please squash this with https://github.com/endlessm/gnome-shell/pull/11/commits/36ba8fdac4801f47861874497df38a8cf8fb8930

cosimoc commented on js/ui/search.js:
```diff
@@ -603,25 +604,51 @@ const SearchResults = new Lang.Class({
+        if (!this._isAnimating) {
+            this._providers.forEach(Lang.bind(this, function (provider) {
+                let results = this._results[provider.id];
+                if (results) {
```
Remove curly braces for single-line if

mariospr replied:
Ok

cosimoc commented on js/ui/workspaceMonitor.js:
```diff
@@ -0,0 +1,68 @@
+        let primaryMonitor = Main.layoutManager.primaryMonitor;
+        this._inFullscreen = primaryMonitor && primaryMonitor.inFullscreen;
+
+        this._appSystem = Shell.AppSystem.get_default();
```
Instead of saving this into a private variable, just call the singleton getter in the only place where it's needed.

mariospr replied:
I'm not sure that's better (I kind of like calling the getter once only for the lifetime of workspaceMonitor), but ok

cosimoc commented on data/Makefile.am:
```diff
@@ -2,7 +2,7 @@ CLEANFILES =
 
 desktopdir=$(datadir)/applications
-desktop_DATA = org.gnome.Shell.desktop gnome-shell-extension-prefs.desktop eos-launch.desktop
+desktop_DATA = org.gnome.Shell.desktop gnome-shell-extension-prefs.desktop eos-launch.desktop eos-link-feedback.desktop
```
If this only ends up being used by Yelp, I'd want to move it to that package.

mariospr replied:
This commit is from @GeorgesStavracas, not sure how it ended up in this PR (should be gone now)

In any case, I agree with you that this should be moved to Yelp

cosimoc commented on src/st/st-entry.c:
```diff
@@ -241,10 +245,10 @@ st_entry_dispose (GObject *object)
   StEntryPrivate *priv = ST_ENTRY_PRIV (entry);
   GdkKeymap *keymap;
 
-  if (priv->entry)
```
Sorry to nitpick, but I think this change should be separated out into another commit. It's not related to this IIRC -- see https://github.com/endlessm/eos-desktop/commit/b27e85340f197d82da5372a1f26baaff27e3f4d1

mariospr replied:
Done

cosimoc commented on js/misc/util.js:
```diff
@@ -57,6 +61,29 @@ function findUrls(str) {
+function findSearchUrls(terms) {
+    let res = [], match;
+    for (let i in terms) {
+        while ((match = _searchUrlRegexp.exec(terms[i]))) {
```
Remove curly braces for single-line condition

mariospr replied:
Ok

cosimoc commented on js/ui/internetSearch.js:
```diff
@@ -0,0 +1,139 @@
+    if (searchedUris.length == 1 && terms.length == 1) {
+        let uri = searchedUris[0];
+        // Ensure all uri has a scheme name
+        if (!GLib.uri_parse_scheme(uri)) {
```
Remove these curlies too

mariospr replied:
Ok

cosimoc commented on js/ui/internetSearch.js:
```diff
@@ -0,0 +1,139 @@
+
+function getInternetSearchProvider() {
+    let browserApp = Util.getBrowserApp();
+    if (browserApp) {
```
Remove these curlies too

mariospr replied:
Ok

cosimoc commented on js/ui/internetSearch.js:
```diff
@@ -0,0 +1,139 @@
+        if (this._networkMonitor.network_available) {
+            let uri = getURIForSearch(terms);
+            let query = terms.join(' ');
+            if (uri) {
```
Remove this and the below curlies too

mariospr replied:
Ok

cosimoc commented on js/ui/internetSearch.js:
```diff
@@ -0,0 +1,139 @@
+
+    launchSearch: function(terms) {
+        this.getInitialResultSet(terms, Lang.bind(this, function(results) {
+            if (results) {
```
Remove these curly braces

mariospr replied:
Ok

cosimoc commented on js/ui/search.js:
```diff
@@ -425,6 +426,11 @@ const SearchResults = new Lang.Class({
         this._registerProvider(new AppDisplay.AppSearchProvider());
+
+        this._internetProvider = InternetSearch.getInternetSearchProvider();
+        if (this._internetProvider) {
```
Remove curly braces

mariospr replied:
Ok

cosimoc commented on src/shell-app-system.c:
```diff
@@ -207,14 +207,6 @@ app_is_stale (ShellApp *app)
-                       gpointer value,
-                       gpointer user_data)
-{
-  return app_is_stale (value);
```
Squash this commit into https://github.com/endlessm/gnome-shell/pull/11/commits/d7372ddaf8e7ee753f2a919c3af7d43afa1d7ce3

cosimoc commented:
@mariospr Also please rebase the branch to master.

mariospr commented:
> @mariospr Also please rebase the branch to master.

Sorry, it's rebased now
